TM1py/Services/CellService.py

```diff
@@ -991,6 +1011,156 @@ def write_through_unbound_process(self, cube_name: str, cellset_as_dict: Dict, i
+        filename = f'{unique_hash}.csv'
+
+        # Upload CSV to TM1 server using ApplicationService:
+        # Create a folder TM1py
+        tm1py_folder = FolderApplication(path='', name='TM1py')
+        if not application_service.exists(path='', application_type='FOLDER', name=tm1py_folder.name):
+            application_service.create(tm1py_folder)
+
+        # Update or create the CSV file inside the TM1py folder
+        application = DocumentApplication(
+            path=tm1py_folder.name,
+            name=filename,
+            content=csv_content.getvalue().encode('utf-8'))
+
+        application_service.create(application=application, private=True, **kwargs)
+        try:
+            # Retrieve the file ID to be used as data source in the TI process. Default folder is .\Externals
+            blob_filename = application_service.get_document(
+                path=tm1py_folder.name,
+                name=filename,
+                private=True).file_id
+
+            # Create a TI process to load CSV  data into the cube
+            process_name = f"}}tm1py.{unique_hash}"
+            process = self._create_blob_write_process(cube_name, process_name, blob_filename,
+                                                      cube_service.get_dimension_names(cube_name), increment,
+                                                      skip_non_updateable, sandbox_name)
+
+            # Call the TI process with result
+            # process_service.create(process)
+            success, status, log_file = process_service.execute_process_with_return(process=process)
+            if not success:
+                if status in ['HasMinorErrors']:
+                    raise TM1pyWritePartialFailureException([status], [log_file], 1)
+                else:
+                    raise TM1pyWriteFailureException([status], [log_file])
+
+        # delete application in TM1
+        finally:
+            application_service.delete(
+                path='TM1py',
+                application_type='DOCUMENT',
+                application_name=filename,
+                private=True)
+
+    def _create_blob_write_process(self, cube_name, dataload_process_name, blob_filename, cube_dimensions, increment,
+                                   skip_non_updateable, sandbox_name):
+        dataload_process = Process(
+            name=dataload_process_name,
+            datasource_type='ASCII',
+            datasource_ascii_header_records=0,
+            datasource_data_source_name_for_server='.\\}Externals\\' + blob_filename,
+            datasource_data_source_name_for_client='.\\}Externals\\' + blob_filename,
+            datasource_ascii_delimiter_char=',',
+            datasource_ascii_decimal_separator='.',
+            datasource_ascii_thousand_separator='',
+            datasource_ascii_quote_character='"')
+        cube_measure = cube_dimensions[-1]
+
+        # Create variables as all String
+        variable_prefix = 'v'
```

Comment:
why not just hard code instead of creating a variable?

Reply:
> I am intrigued by seeing such a big difference in use_ti vs use_blob. In my experience, the creation of the previous unbound TI with the CellPutN statements, did not add a significant processing time. Any insight on what might be creating this improvement.

Here is my hypothesis:
The TM1 server does some kind of "scanning" when a process is created. That operation is expensive in terms of performance and the cost scales linear with the number of lines in the process.
https://github.com/cubewise-code/tm1py/issues/774#issuecomment-1258088290

In the old approach, we would create a process with 1 million lines. Now we always create a process with 15 lines or so. Even if we write 1 million elements.